lib/src/facts/resolvers/augeasversion_resolver.cc

```diff
@@ -0,0 +1,59 @@
+#include <internal/facts/resolvers/augeasversion_resolver.hpp>
+#include <internal/util/regex.hpp>
+#include <facter/facts/collection.hpp>
+#include <facter/facts/fact.hpp>
+#include <facter/facts/scalar_value.hpp>
+#include <facter/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+
+using namespace std;
+using namespace facter::util;
+
+namespace facter { namespace facts { namespace resolvers {
+
+    augeasversion_resolver::augeasversion_resolver() :
+        resolver(
+            "augeasversion",
+            {
+                fact::augeasversion
+            })
+    {
+    }
+
+    string augeasversion_resolver::get_version()
+    {
+        string augtool = [] {
```

Comment:
This is a pattern I used in https://github.com/puppetlabs/facter/blob/master/lib/src/facts/linux/virtualization_resolver.cc#L105.